### PR TITLE
Docs for `variant="spectral"` on simplices

### PR DESCRIPTION
**Pull request by pbrubeck**
# Description
These have to go together with https://github.com/firedrakeproject/tsfc/pull/301

### CODE REVIEW

File: docs/source/variational-problems.rst
Comment (pbrubeck) [inline docs/source/variational-problems.rst:666]:
```suggestion
.. _recursivenodes: https://tisaac.gitlab.io/recursivenodes/
```

**Review by JDBetteridge — CHANGES_REQUESTED** [review]
This is fine, but dependent on https://github.com/firedrakeproject/tsfc/pull/301

**Review by connorjward — COMMENTED** [review]
I would quite like to wait for the docs build to pass so we can review the artefact before merging. In any case there is little point in merging before then because no one will see the changes.

**Review by connorjward — APPROVED** [review]
I've checked the build artefact and this is good. Thanks!

**Pull request merged**
